utils_cv/detection/dataset.py

```diff
@@ -386,15 +428,35 @@ def im_gen() -> torch.Tensor:
             print(f"Transformations applied on {self.im_paths[idx]}:")
             [print(transform) for transform in self.transforms.transforms]
 
-    def _get_random_anno(
-        self
-    ) -> Tuple[List[AnnotationBbox], Union[str, Path]]:
+    def _get_binary_mask(self, idx: int) -> Union[np.ndarray, None]:
+        """ Return binary masks for objects in the mask image. """
+        binary_masks = None
+        if self.mask_paths:
+            if self.mask_paths[idx] is not None:
+                binary_masks = binarise_mask(Image.open(self.mask_paths[idx]))
+            else:
+                # for the tiny bounding box in _read_annos(), make the mask to
+                # be the whole box
+                mask = np.zeros(
+                    Image.open(self.im_paths[idx]).size[::-1],
+                    dtype=np.uint8
+                )
+                mask[1:6, 1:6] = 1
+                binary_masks = binarise_mask(mask)
+
+        return binary_masks
+
+    def _get_random_anno(self) -> Tuple:
         """ Get random annotation and corresponding image
 
         Returns a list of annotations and the image path
         """
-        idx = randrange(len(self.anno_paths))
-        return self.anno_bboxes[idx], self.im_paths[idx]
+        idx = random.randrange(len(self.im_paths))
+
+        # get mask if any
+        mask = self._get_binary_mask(idx)
```

Comment:
Could just add this on the same line below, ie.
return self.anno_bboxes[idx], self.im_paths[idx], self._get_binary_mask(idx)

Reply:
Done